lib/ansible/modules/network/ios/ios_vlan.py

```diff
@@ -211,38 +210,52 @@ def map_params_to_obj(module):
     return obj
 
 
-def map_config_to_obj(module):
-    output = run_commands(module, ['show vlan'])
-    lines = output[0].strip().splitlines()[2:-1]
-
-    if not lines:
-        return list()
-
-    objs = list()
-
-    for l in lines:
-        splitted_line = l.strip().replace(",", "").split()
-        if splitted_line == []:
-            break
-        obj = {}
-        obj['vlan_id'] = splitted_line[0]
-        obj['name'] = splitted_line[1]
-        obj['state'] = splitted_line[2]
-
-        if obj['state'] == 'suspended':
-            obj['state'] = 'suspend'
-
-        obj['interfaces'] = []
-        if len(splitted_line) > 3:
-            interface = []
-            for i in range(3, len(splitted_line)):
-                interface.append(splitted_line[i].replace('Gi', 'GigabitEthernet'))
-            obj['interfaces'].extend(interface)
-        objs.append(obj)
-
+def logical_rows(out):
```

Comment:
It's not the most ideal code that I wrote, so if you feel you could improve things, especially function and variable naming — go for it :)

For example, function names normally designate actions and start with verbs (like `get_logical_rows()` or `extract_logical_rows()`), as opposed to variables, which are typically names of some things / nouns.

Keep in mind that I did a very quick sketch, so I might've missed some things, I rely on you here to check whether it fixes the issue and doesn't cause new problems.